shepherd_bms/include/datastructs.h

```diff
@@ -122,4 +122,77 @@ struct AccumulatorData_t
 
 };
 
+/**
+ * @brief Represents individual BMS states
+ */
+typedef enum
+{
+    BOOT_STATE,       //State when BMS first starts up, used to initialize everything that needs configuring
+    READY_STATE,      //State when car is not on/BMS is not really doing anything
+    CHARGING_STATE,   //State when car is on and is charging (Filling battery)
+    FAULTED_STATE,    //State when BMS has detected a catastrophic fault and we need to hault operations
+    NUM_STATES
+
+} BMSState_t;
+
+/**
+ * @brief Represents the state of the BMS fault timers
+ */
+typedef enum
+{
+	BEFORE_TIMER_START,
+	DURING_FAULT_EVAL
+
+} FaultEvalState;
+
+/**
+ * @brief Represents fault evaluation operators
+ */
+typedef enum
+{
+    GT = 0, // fault if {data} greater than {threshold}
```

Comment:
nit: Since this is an enum, you don't need to explicitly set the values equal to something, its only really nice to do that if the values are nonsequential or start at a weird value